nixops/script_defs.py

```diff
@@ -782,157 +818,166 @@ def scp_loc(user, ssh_name, remote, loc):
+            if flags[n] == "-i":
+                new_flags.extend(["-o", "IdentityFile=" + flags[n + 1]])
+                n = n + 2
+            elif flags[n] == "-p":
+                new_flags.extend(["-p", flags[n + 1]])
+                n = n + 2
+            elif flags[n] == "-o":
+                new_flags.extend(["-o", flags[n + 1]])
+                n = n + 2
+            else:
+                raise Exception(
+                    "don't know how to pass SSH flag ‘{0}’ to sshfs".format(flags[n])
+                )
 
-    for o in args.sshfs_option or []:
-        new_flags.extend(["-o", o])
+        for o in args.sshfs_option or []:
+            new_flags.extend(["-o", o])
 
-    # Note: sshfs will go into the background when it has finished
-    # setting up, so we can safely delete the SSH identity file
-    # afterwards.
-    res = subprocess.call(
-        ["sshfs", username + "@" + ssh_name + ":" + remote_path, args.destination]
-        + new_flags
-    )
-    sys.exit(res)
+        # Note: sshfs will go into the background when it has finished
+        # setting up, so we can safely delete the SSH identity file
+        # afterwards.
+        res = subprocess.call(
+            ["sshfs", username + "@" + ssh_name + ":" + remote_path, args.destination]
+            + new_flags
+        )
+        sys.exit(res)
 
 
 def op_show_option(args):
-    depl = open_deployment(args)
-    if args.include_physical:
-        depl.evaluate()
-    sys.stdout.write(
-        depl.evaluate_option_value(
-            args.machine,
-            args.option,
-            json=args.json,
-            xml=args.xml,
-            include_physical=args.include_physical,
+    with deployment(args) as depl:
+        if args.include_physical:
+            depl.evaluate()
+        sys.stdout.write(
+            depl.evaluate_option_value(
+                args.machine,
+                args.option,
+                json=args.json,
+                xml=args.xml,
+                include_physical=args.include_physical,
+            )
         )
-    )
 
 
-def check_rollback_enabled(args):
-    depl = open_deployment(args)
-    if not depl.rollback_enabled:
-        raise Exception(
-            "rollback is not enabled for this network; please set ‘network.enableRollback’ to ‘true’ and redeploy"
-        )
-    return depl
+@contextlib.contextmanager
+def with_rollback_enabled(args):
+    with deployment(args) as depl:
+        if not depl.rollback_enabled:
+            raise Exception(
+                "rollback is not enabled for this network; please set ‘network.enableRollback’ to ‘true’ and redeploy"
+            )
+        yield depl
 
 
 def op_list_generations(args):
-    depl = check_rollback_enabled(args)
-    if (
-        subprocess.call(["nix-env", "-p", depl.get_profile(), "--list-generations"])
-        != 0
-    ):
-        raise Exception("nix-env --list-generations failed")
+    with with_rollback_enabled(args) as depl:
```

Comment:
Maybe this is nitpicking but I feel that `with with_` is a bit strange.

Reply:
Oh yeah, totally. I don't really like the names of these context managers. Maybe you have a different suggestion?

Reply:
Maybe invert the naming and call it `deployment_with_rollback`? It also makes things more discoverable.

Reply:
Fixed!